pkg/grep.go

```diff
@@ -0,0 +1,89 @@
+package pkg
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Grep struct {
+	kvs []*grepKV
+}
+
+func NewGrep(expr string) (*Grep, error) {
+	var kvs []*grepKV
+
+	if len(expr) != 0 {
+		for _, kv := range strings.Split(expr, ",") {
+			pair := strings.Split(kv, "=")
+			if len(pair) != 2 {
+				return nil, fmt.Errorf("grep: invalid key=value pair %q in expression %q", kv, expr)
+			}
+			k := pair[0]
```

Comment:
Would it make sense to `strings.TrimSpace()` for `k` and `v` here, because there's check for first character below and that would fail if there's space. I could imagine e.g. a grep string: `--grep='foo=bar, !baz=foo, grep=patter'` and in this case I think the check below would miss `k=' !baz'`.

Reply:
Done.